### PR TITLE
Pal 705 fix perf test errors

### PR DESCRIPTION
**Pull request by pd104923**

### CODE REVIEW

File: example-model/src/main/java/uk/gov/gchq/palisade/example/model/client/ExampleSimpleClient.java [hunk 1]
```diff
@@ -38,18 +40,34 @@ public ExampleSimpleClient(final PalisadeClient palisadeClient, final DataClient
...
         }
-        return super.read(absoluteFile, userId, purpose);
+        // Get a file:$fileName URI
+        String resourceId = file.toURI().toString();
+
+        // Check that the resourceId ending is consistent with the fileName
+        // ie. "../some/directory/" should still have a trailing slash "file:/root/some/directory/"
+        // fileName could have been a DOS path
+        // resourceId is a URI, so only a UNIX path
+        if ((fileName.endsWith("/") || fileName.endsWith("\\")) && !resourceId.endsWith("/")) {
```
Comment (developer01189998819991197253) [inline example-model/src/main/java/uk/gov/gchq/palisade/example/model/client/ExampleSimpleClient.java:67]: is there a better way to do this? maybe converting the fileName to a File object and checking File.seperator?

File: performance/src/main/java/uk/gov/gchq/palisade/example/perf/actions/CreateAction.java [hunk 2]
```diff
@@ -94,14 +95,14 @@ private static void copyDir(final Path src, final Path dest) {
     private boolean createWithPolicyDataset(final PerfFileSet fileSet) {
         ExecutorService tasks = Executors.newFixedThreadPool(NUMBER_OF_THREADS, Util.createDaemonThreadFactory());
 
-        Path smallFile = fileSet.smallFile;
-        Path largeFile = fileSet.largeFile;
-        Path manyDir = fileSet.manyDir;
-        Path masterCopy = manyDir.resolve(String.format(PerfUtils.MANY_DUP_DIR_FORMAT, 0));
+        String smallFile = fileSet.smallFile;
```
Comment (developer01189998819991197253) [inline performance/src/main/java/uk/gov/gchq/palisade/example/perf/actions/CreateAction.java:98]: Why the change of types just to have to create them as new Path(var) below?

File: example-model/src/main/java/uk/gov/gchq/palisade/example/model/client/ExampleSimpleClient.java (hunk 1 above)
Comment (pd104923) [reply example-model/src/main/java/uk/gov/gchq/palisade/example/model/client/ExampleSimpleClient.java:67]: We need to check if the original fileName is a directory or not and then check that against the resourceId value that has been created from the `new File(filename)`.
Originally this check was not being done and we only used the `getCononicalFile()` value which dropped the final "/" when it was a directory which caused the perf trials to fail.

File: performance/src/main/java/uk/gov/gchq/palisade/example/perf/actions/CreateAction.java (hunk 2 above)
Comment (pd104923) [reply performance/src/main/java/uk/gov/gchq/palisade/example/perf/actions/CreateAction.java:98]: The new types are coming from the PerfFileSet and these are being separated out for ease of use. The `Path.of()` is only being done for the manyDir.
The manyDir could be converted to a `Path` from the `String` value but the other two are used to create in the `new File(var)`
